swarm/api/services/run_state.py

```
@@ -174,24 +174,37 @@ def list_runs(self, limit: int = 20) -> List[Dict[str, Any]]:
+                state = self._cache[run_id]
+            else:
+                state_path = run_dir / "run_state.json"
+                if not state_path.exists():
+                    continue
+
+                try:
+                    state = json.loads(state_path.read_text(encoding="utf-8"))
+                    # Note: We don't populate cache here to avoid evicting frequently used items
+                    # for a one-off listing. But if we wanted to warm the cache, we could:
+                    # self._cache[run_id] = state
+                except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)